library/rendering/item_rendering/src/main/java/org/quiltmc/qsl/rendering/item/mixin/client/ItemRendererMixin.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.rendering.item.mixin.client;
+
+import org.jetbrains.annotations.Nullable;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.Unique;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.Redirect;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import org.spongepowered.asm.mixin.injection.callback.LocalCapture;
+
+import net.minecraft.client.font.TextRenderer;
+import net.minecraft.client.render.item.ItemRenderer;
+import net.minecraft.client.util.math.MatrixStack;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+
+import org.quiltmc.qsl.rendering.item.api.client.CooldownOverlayRenderer;
+import org.quiltmc.qsl.rendering.item.api.client.CountLabelRenderer;
+import org.quiltmc.qsl.rendering.item.api.client.ItemBarRenderer;
+
+@Mixin(ItemRenderer.class)
+public abstract class ItemRendererMixin {
+	@Unique private final ThreadLocal<MatrixStack> quilt$matrices = ThreadLocal.withInitial(MatrixStack::new);
```

Comment:
Do we really need to use ThreadLocal here? I don't think this is accessed by any other thread, other than the Render Thread.

Reply:
Its good practice. Other mods might also render twice like Replay mod (Not sure)